api/client/cli.go

```diff
@@ -112,8 +112,13 @@ func NewDockerCli(in io.ReadCloser, out, err io.Writer, clientFlags *cli.ClientF
 			return errors.New("Please specify only one -H")
 		}
 
+		defaultHost := opts.DefaultTCPHost
+		if clientFlags.Common.TLSOptions != nil {
+			defaultHost = opts.DefaultTLSHost
+		}
+
 		var e error
-		if hosts[0], e = opts.ParseHost(hosts[0]); e != nil {
+		if hosts[0], e = opts.ParseHost(defaultHost, hosts[0]); e != nil {
```

Comment:
@duglin  passing the `defaultHost` to `opts` package is to let `ParseDockerDaemonHost`(https://github.com/coolljt0725/docker/blob/default_tls_host/pkg/parsers/parsers.go#L20) know what the `defaultHost` is, because `defaultHost` could be `/var/run/docker.sock` or `localhost:2376`(for tls)